tests/console_test.go

```diff
@@ -147,16 +147,6 @@ var _ = Describe("[rfe_id:127][posneg:negative][crit:medium][vendor:cnv-qe@redha
 				ExpectConsoleOutput(vmi, "login")
 			}, 220)
 
-			It("[test_id:1592]should wait until the virtual machine is in running state and return a stream interface", func() {
```

Comment:
Redundant but explicit.

Reply:
Test 1588 is waiting for the VMI to be launched and then proceeding to expect console output. Here we expect the console right after we create the VMI. Meaning the VMI is for 99.99% not ready.

I am not saying we keep it( I can be persuaded). I personally didn't go over all tests to see if we test this situation. What I know is that if the test exists it certainly doesn't need to stay that way. But this test needs to stay that way and therefore provides a better guruantee that we test such scenerio.

Reply:
I will try to be more clear, please be patient with me.

> Do we have a requirement/expectation to connect to the console before it is launched?

There is an expectation that this will not return an error and we will **eventually** connect to the vm. (At least I understand this test like this)

Ad. `test_id1593`: Here we are using `nodeAffinity` to non-existent node causing the VMI to never be launched. Therefore this can't ever succed.

>If I understand you correctly, the test I suggest to remove here is /raceful/: there may be a case where the VM is created; qemu is not yet running, but the test already attempts to connect to its console. To me, this is another reason to drop this test. But maybe I do not understand your suggestion.

I don't agree because I think the test is creating the expectation that we eventually connect to VMI(even when the qemu is not yet running).

It is totally possible I am also misunderstanding the intent and expected behavior.

Reply:
I have have tons of patience for a serious review like yours. I hope it shows.

> I don't agree because I think the test is creating the expectation that we eventually connect to VMI(even when the qemu is not yet running).

Ah, I now see what you mean. Due to a race, this test may degenerate to one that first waits for the VM to be up, and only then connects to the console. But typically it does not happen and we get some kind of evidence that serial console blocks until vm start-up. I wish there was a way to define a VM, request a console connection, and only then have the VM started. I don't like this test in its current form, but let's keep it for now.

Reply:
> I wish there was a way to define a VM, request a console connection, and only then have the VM started

I think something like this will soon be possible.